Dev: add option to disable service worker

NachoLZ opened the pull request:
### The Problem

When using middleware for authentication (e.g., authentik), the service worker's caching of the root page (`/`) can interfere with the authentication flow. If a user's session expires, the cached page is served, preventing the middleware from redirecting to the login page. This is particularly problematic in environments where the app and server are on the same domain.

This change addresses the issue reported in [GitHub Issue #986](https://github.com/Stremio/stremio-web/issues/986).

### The Solution

A `SERVICE_WORKER_DISABLED` environment variable has been added. When this variable is set to `'true'`, the application will skip the registration of the service worker. This allows authentication middleware to function as expected without interference from the cached root page.

### How to Use

To disable the service worker, set the following environment variable in your build environment:

`SERVICE_WORKER_DISABLED=true`

By default, if the variable is not set or is set to `false`, the service worker will be registered as usual.

### Code of Conduct
- [x] I agree

NachoLZ committed:
feat: add option to disable service worker

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Stremio/stremio-web?pullRequest=1018) <br/>All committers have signed the CLA.

tymmesyde commented on src/index.js:
```diff
@@ -36,7 +36,7 @@ i18n
 root.render(<App />);
 
-if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
+if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator && process.env.SERVICE_WORKER_DISABLED !== 'true') {
```
in this case if you set `SERVICE_WORKER_DISABLED` to `true` in webpack.config.js, it will not disable it because it's parsed as boolean instead of a string
```suggestion
if (process.env.NODE_ENV === 'production' && process.env.SERVICE_WORKER_DISABLED !== 'true' && process.env.SERVICE_WORKER_DISABLED !== true && 'serviceWorker' in navigator) {
```

NachoLZ replied:
Great catch, thank you! I
I've committed your suggestion.

NachoLZ committed:
fix: Handle boolean value for SERVICE_WORKER_DISABLED

The `webpack.EnvironmentPlugin` provides the default value for `SERVICE_WORKER_DISABLED` as a boolean (`false`).

The previous implementation only checked for the string `'true'`, which would fail to correctly identify the boolean `true` case, causing the feature to not work as intended when the variable was set without being explicitly a string.

This commit updates the conditional check to handle both the boolean `true` and the string `'true'` to ensure the service worker is reliably disabled.

Co-authored-by: Tim <tymmesyde@gmail.com>

tymmesyde commented:
Thank you!

xam1dullo reviewed (commented):
ads
